chains/evm/deployment/utils/changesets/mcms.go

```diff
@@ -0,0 +1,164 @@
+package changesets
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	cldf_deployment "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	mcmsevmsdk "github.com/smartcontractkit/mcms/sdk/evm"
+	"github.com/smartcontractkit/mcms/types"
+	mcms_types "github.com/smartcontractkit/mcms/types"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/utils"
+	datastore_utils "github.com/smartcontractkit/chainlink-ccip/chains/evm/deployment/utils/datastore"
+)
+
+type MCMSRole string
+
+var (
+	BypasserManyChainMultisig  datastore.ContractType = "BypasserManyChainMultiSig"
+	CancellerManyChainMultisig datastore.ContractType = "CancellerManyChainMultiSig"
+	ProposerManyChainMultisig  datastore.ContractType = "ProposerManyChainMultiSig"
+	ManyChainMultisig          datastore.ContractType = "ManyChainMultiSig"
+	RBACTimelock               datastore.ContractType = "RBACTimelock"
+	CallProxy                  datastore.ContractType = "CallProxy"
+
+	// roles
+	ProposerRole  MCMSRole = "PROPOSER"
+	BypasserRole  MCMSRole = "BYPASSER"
+	CancellerRole MCMSRole = "CANCELLER"
+)
+
+func (r MCMSRole) String() string {
+	return string(r)
+}
+
+type WithMCMS interface {
+	MCMSConfig() *MCMSConfig
+	TimelockAddressQualifier() string
+	MCMSAddressQualifier() string
+}
+
+func ResolveMCMSParams[CFG WithMCMS](e cldf_deployment.Environment, cfg CFG) (MCMSParams, error) {
+	if cfg.MCMSConfig() == nil {
+		return MCMSParams{}, nil
+	}
+	// map of chain selector to timelock address
+	timelockAddresses := make(map[types.ChainSelector]string)
+	chainMetadata := make(map[types.ChainSelector]mcms_types.ChainMetadata)
+	for sel := range e.BlockChains.EVMChains() {
+		// find and format the MCMS related addresses
+		timelockAddrs, err := datastore_utils.FindAndFormatEachRef(e.DataStore, []datastore.AddressRef{
+			{
```

Comment:
We should just take the full ref from user input (i.e. Timelock Ref), same as the MCM contract. We then replace the selector in the ref with the target chain selector, as we should override the correct value there despite whether or not the user set anything for it.

Maybe we can add a common func for this in the datastore pkg: updateRefWithChainSelector(ref datastore.AddressRef, chainSelector uint64)